core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -759,13 +773,20 @@ class ZkUtils(zkClientWrap: ZooKeeperClientWrapper,
   def getPartitionLeaderAndIsrForTopics(topicAndPartitions: Set[TopicAndPartition]): mutable.Map[TopicAndPartition, LeaderIsrAndControllerEpoch] = {
```

Comment:
We have getTopicPartitionStates, getTopicPartitionState methods in KafkaZkClient. Can we use these methods, so that we can avoid dependency on ZKUtils methods?

Reply:
This is in ZkUtils...

Reply:
I was thinking callers of this method can call KafkaZkClient.getTopicPartitionStates method.
This also used in few test classes. This can be done later.

Reply:
Agreed, as in the other comment, let's do test cases last as it will be clearer what needs to be done then.